src/utils.cpp

```diff
@@ -1077,13 +1077,19 @@ std::string str_join(const std::vector<std::string> &list,
   return str;
 }
 
-bool is_numeric(const std::string &s)
+bool is_numeric(const std::string &s, bool &is_signed)
```

Comment:
Catching exceptions can be horribly slow so we should try to avoid their use in happy-path parsing. This particular function has been on my TODO list for a long time but I've never gotten around to changing it.

Reply:
We could use a regex to determine if the string is a hex and also look at the first character to see if it's a "-" to see if we need a signed int.

Also, I realized I need to fix some of the types for `get_int_literal`.

All that being said, I feel like perf improvements to functions not called multiple times in a hot path shouldn't block us from fixing bugs like this. We can always do follow-up PRs for speed-ups. I'll get off my soap box now.

Reply:
Ok, I made some changes. It's a more complicated version so TBH I'm not sure if I made it better but at least it doesn't rely on exceptions to determine if the string is signed or not. Let me know what you think.

I didn't tackle the fact that `get_int_literal` only returns int64_t - mostly because it was more complicated and I wasn't sure if we want to do something similar to what I have now for `get_int_from_str` where I return a variant. Also didn't want to bloat the PR.

Reply:
> All that being said, I feel like perf improvements to functions not called multiple times in a hot path shouldn't block us from fixing bugs like this. We can always do follow-up PRs for speed-ups. I'll get off my soap box now.

I agree with this. If it's not being called many times I'd prioritize correctness and maintainability instead. Also, I kinda wonder if running a regex is faster than catching an exception. At least we can make the common case not throw - then there is no cost at all.